code/test-code.sh

```diff
@@ -46,26 +49,25 @@ mkdir -p $CODE
 pushd $CODE
 
 npm install --save neo4j-driver 
-sed -e "s/<BOLTPORT>/$BOLTPORT/g" -e "s/<HOST>/$HOST/g" -e "s/mUser/$USERNAME/g" -e "s/s3cr3t/$PASSWORD/g" $TARGET/code/javascript/example.js > example.js
+sed -e "s/<BOLTPORT>/$BOLTPORT/g" -e "s/<HOST>/$HOST/g" -e "s/<USERNAME>/$USERNAME/g" -e "s/<PASSWORD>/$PASSWORD/g" $TARGET/code/javascript/example.js > example.js
+
 node example.js | grep "$EXPECT" || echo "JAVASCRIPT FAIL"
 
 pip install neo4j-driver
-sed -e "s/<BOLTPORT>/$BOLTPORT/g" -e "s/<HOST>/$HOST/g" -e "s/mUser/$USERNAME/g" -e "s/s3cr3t/$PASSWORD/g" $TARGET/code/python/example.py > example.py
+sed -e "s/<BOLTPORT>/$BOLTPORT/g" -e "s/<HOST>/$HOST/g" -e "s/<USERNAME>/$USERNAME/g" -e "s/<PASSWORD>/$PASSWORD/g" $TARGET/code/python/example.py > example.py
 python example.py | grep "$EXPECT" || echo "PYTHON FAIL"
 
 curl -sOL https://repo1.maven.org/maven2/org/reactivestreams/reactive-streams/${RX_VERSION}/reactive-streams-${RX_VERSION}.jar
 curl -sOL https://repo1.maven.org/maven2/org/neo4j/driver/neo4j-java-driver/${JAVA_DRIVER_VERSION}/neo4j-java-driver-${JAVA_DRIVER_VERSION}.jar
 
-sed -e "s/<BOLTPORT>/$BOLTPORT/g" -e "s/<HOST>/$HOST/g" -e "s/mUser/$USERNAME/g" -e "s/s3cr3t/$PASSWORD/g" $TARGET/code/java/Example.java > Example.java
+sed -e "s/<BOLTPORT>/$BOLTPORT/g" -e "s/<HOST>/$HOST/g" -e "s/<USERNAME>/$USERNAME/g" -e "s/<PASSWORD>/$PASSWORD/g" $TARGET/code/java/Example.java > Example.java
 
 javac -cp neo4j-java-driver-${JAVA_DRIVER_VERSION}.jar Example.java
 java -cp neo4j-java-driver-${JAVA_DRIVER_VERSION}.jar:reactive-streams-${RX_VERSION}.jar:. Example  | grep "$EXPECT" || echo "JAVA FAIL"
 
-sed -e "s/<BOLTPORT>/$BOLTPORT/g" -e "s/<HOST>/$HOST/g" -e "s/mUser/$USERNAME/g" -e "s/s3cr3t/$PASSWORD/g" $TARGET/code/go/example.go > example.go
+sed -e "s/<BOLTPORT>/$BOLTPORT/g" -e "s/<HOST>/$HOST/g" -e "s/<USERNAME>/$USERNAME/g" -e "s/<PASSWORD>/$PASSWORD/g" $TARGET/code/go/example.go > example.go
 go mod init main
 go run example.go | grep "$EXPECT" || echo "GO FAIL"
 
 popd
 
-echo rm -rf $TMP
```

Comment:
when how is the docker container and code cleaned up?

Reply:
there is a `--rm` flag and a trap to kill the container upon exit (see lines 28 and 29)
I removed the files deletion as it makes debugging harder when something goes wrong